### PR TITLE
Optimized N-Queen Problem using JAVA 

### PR DESCRIPTION
**Pull request by Shreyash41**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description

Implementing Optimized N-Queens Problem using Java Language.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#2210

### CODE REVIEW

**Comment by atarax665** [comment]
Thanks for making your Pull Request!

PR would be reviewed after the start of the Coding Period. Thanks!

File: Java/cp/N_Queens_Problem_Efficient.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+         1  0  0  0
+         0  0  0  1
+         0  1  0  0
+ Input : N = 5
+         1  0  0  0  0
+         0  0  0  1  0
+         0  1  0  0  0
+         0  0  0  0  1
+         0  0  1  0  0
+ */
```
Comment (HarshCasper) [inline Java/cp/N_Queens_Problem_Efficient.java]: Add a new line at the EOF

File: Java/cp/N_Queens_Problem_Efficient.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
+package com.company;
```
Comment (HarshCasper) [inline Java/cp/N_Queens_Problem_Efficient.java]: Remove Packages

File: Java/cp/N_Queens_Problem_Efficient.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+/* Java program to solve N Queen Problem
+using backtracking in an Optimization Method*/
+import java.util.*;
+
+class anaGram
```
Comment (HarshCasper) [inline Java/cp/N_Queens_Problem_Efficient.java]: Can you use more meaningful class names?

File: Java/cp/N_Queens_Problem_Efficient.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+{
+    static int N;
+    static int []neg = new int[30];// to store negative indices
+    static int []dia = new int[30];// queen placed in right diagonal or not(row+column)
+    static int []col = new int[30];// column array
```
Comment (HarshCasper) [inline Java/cp/N_Queens_Problem_Efficient.java]: Use more meaningful variable names and use comments above the line of the code

File: Java/cp/N_Queens_Problem_Efficient.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    static void printSolution(int onBoard[][])
+    {
+        for (int i = 0; i < N; i++)
+        {
+            for (int j = 0; j < N; j++)
+                System.out.printf(" %d ", onBoard[i][j]);
+            System.out.printf("\n");
+        }
+    }
+    static boolean solveUtil(int onBoard[][], int colo)
```
Comment (HarshCasper) [inline Java/cp/N_Queens_Problem_Efficient.java:33]: Add a function description

File: Java/cp/N_Queens_Problem_Efficient.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+using backtracking in an Optimization Method*/
+import java.util.*;
+
+class anaGram
+{
+    static int N;
+    static int []neg = new int[30];// to store negative indices
+    static int []dia = new int[30];// queen placed in right diagonal or not(row+column)
+    static int []col = new int[30];// column array
+    static void printSolution(int onBoard[][])
```
Comment (HarshCasper) [inline Java/cp/N_Queens_Problem_Efficient.java:17]: Add a function description

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**